metadata_templates.py

```diff
@@ -64,10 +77,82 @@ def metadata_templates_collection_detach(rule_args, callback, rei):
     result = mt_detach(callback, logical_path, 'collection', schema, type)
 
 
+def metadata_templates_data_object_gather(rule_args, callback, rei):
+    # Export/Collapse/Rasterize/Gather/Dump (logical_path, recursive, schemas_string)
+    # - Find all associated schemas
+    # - Recursive gathers schemas on all parents up to root
+
+    logical_path = rule_args[0]
+    recursive = rule_args[1]
+
+    # split logical path
+    collection_name, data_name = logical_path.rsplit("/", 1)
+
+    # get all schema locations attached to this data object
+    schemas = []
+    for schema, thetype in Query(callback,
+                        "META_DATA_ATTR_VALUE, META_DATA_ATTR_UNITS",
+                        "COLL_NAME = '{}' and DATA_NAME = '{}' and META_DATA_ATTR_NAME = '{}'".format(collection_name, data_name, MT_NAMESPACE)):
+#        callback.writeLine('serverLog','{} {}'.format(thetype, schema))
+        if thetype == 'url':
+            try:
+                # get the schema content from the location
+                r = requests.get(schema)
+                # convert to json object
+                j = json.loads(r.content)
+                # add to schemas array
+                schemas.append(j)
+            except Exception as e:
+                callback.writeLine('serverLog', '{}'.format(type(e)))
```

Comment:
We catch the exception and log it, but then we allow the rule to continue? Is that intentional?